home/tests/test_broken_links_clean_up.py

```diff
@@ -0,0 +1,179 @@
+    def test_content_page_with_deleted_related_page(self) -> None:
+        """
+        If related_page is deleted we still get two objects but one will be null the one that is deleted
+        """
+        output = StringIO()
+
+        self_help = PageBuilder.build_cp(
+            parent=self.ha_menu,
+            slug="self-help",
+            title="self-help",
+            bodies=[WABody("self-help", [WABlk("*Self-help programs*")])],
+        )
+        index = PageBuilder.build_cpi(self.health_info, "iddex-page", "Index Page")
+
+        self_help_rp = Page.objects.get(pk=self_help.id)
+        PageBuilder.link_related(self.health_info, [self_help, index])
+
+        index.delete()
+        related_pages = [rp.value for rp in self.health_info.related_pages]
+        call_command("broken_links_clean_up", stdout=output)
+
+        assert len(self.health_info.related_pages) == 2
+        assert related_pages[1] is None
+        assert related_pages == [self_help_rp, None]
+        assert (
+            output.getvalue().split("\n")[0].strip()
+            == f"Content Page: {self.health_info.id} with non existing related page"
+        )
+        assert output.getvalue() == (
+            f"Content Page: {self.health_info.id} with non existing related page \nSuccessfully retrieve broken links\n"
+        )
+
+    def test_content_page_with_a_go_to_button(self) -> None:
+        """
+        If page linked to button to go to page is deleted we still get two objects
+        """
+        output = StringIO()
+
+        index = PageBuilder.build_cpi(self.health_info, "iddex-page", "Index Page")
+
+        self_help = PageBuilder.build_cp(
+            parent=self.ha_menu,
+            slug="self-help",
+            title="self-help",
+            bodies=[
+                WABody(
+                    "self-help",
+                    [
+                        WABlk(
+                            "*Self-help programs*",
+                            buttons=[
+                                NextBtn("Next message button"),
+                                PageBtn("Import Export", page=index),
+                            ],
+                        )
+                    ],
+                )
+            ],
+        )
+
+        call_command("broken_links_clean_up", stdout=output)
+        orig_self_help_btn = self_help.whatsapp_body._raw_data[0]["value"].get(
+            "buttons"
+        )
+        assert len(orig_self_help_btn) == 2
+
+        index.delete()
+        del_self_help_btn = self_help.whatsapp_body._raw_data[0]["value"].get("buttons")
+
+        assert len(orig_self_help_btn) == 2
+        assert (
+            orig_self_help_btn[0]["value"]["title"]
+            == del_self_help_btn[0]["value"]["title"]
+        )
+        assert (
+            orig_self_help_btn[1]["value"]["page"]
+            == del_self_help_btn[1]["value"]["page"]
+        )
+        assert output.getvalue().strip() == "Successfully retrieve broken links"
```

Comment:
This makes me think that this isn't working as expected? The output isn't listing the buttons pointing to deleted pages? But it might just be an issue with the test, it seems like we're calling the command before we delete the page, so if we call the command afterwards, then we should get the output we expect.

Reply:
Oh yes thanks i see my mistake here.